### PR TITLE
Fixes #15552: Wrong path in shared-files API

### PR DESCRIPTION
**Pull request by ArmandSy**
https://issues.rudder.io/issues/15552

### CODE REVIEW

File: relay/sources/relayd/src/shared_files.rs [hunk 1]
```diff
@@ -240,7 +240,11 @@ pub fn file_writer(buf: &mut warp::body::FullBody, path: &str) {
     buf.by_ref().reader().consume(1); // skip the line feed
     buf.reader().read_to_end(&mut myvec).unwrap();
 
-    fs::write(format!("shared-files/{}", path), myvec).expect("Unable to write file");
+    fs::write(
+        format!("/var/rudder/configuration-repository/shared-files/{}", path),
```
Comment (amousset) [inline relay/sources/relayd/src/shared_files.rs]: We should not use hardcoded paths

File: relay/sources/relayd/src/shared_files.rs [hunk 2]
```diff
@@ -292,24 +296,47 @@ pub fn put_handler(
...
     }
 
-    let _ = fs::create_dir_all(format!("shared-files/{}/{}/", target_uuid, source_uuid)); // on cree les folders s'ils existent pas
-                                                                                          //fs::create_dir_all(format!("/var/rudder/configuration-repository/shared-files/{}/{}/", target_uuid, source_uuid)); // real path
+    let _ = fs::create_dir_all(format!(
+        "{}shared-files/{}/{}/",
+        job_config.cfg.shared_files_path.path.to_str().unwrap(),
+        target_uuid,
+        source_uuid
+    )); // on cree les folders s'ils existent pas
```
Comment (amousset) [inline relay/sources/relayd/src/shared_files.rs]: please use english for comments

File: relay/sources/relayd/src/shared_files.rs [hunk 3]
```diff
@@ -292,24 +296,47 @@ pub fn put_handler(
...
-pub fn metadata_hash_checker(path: String, hash: String) -> hyper::StatusCode {
-    if !Path::new(&format!("shared-files/{}.metadata", path)).exists() {
+pub fn metadata_hash_checker(
+    path: String,
+    hash: String,
+    job_config: Arc<JobConfig>,
+) -> hyper::StatusCode {
+    if !Path::new(&format!(
+        "{}shared-files/{}.metadata",
+        job_config.cfg.shared_files_path.path.to_str().unwrap(),
```
Comment (amousset) [inline relay/sources/relayd/src/shared_files.rs]: you should use path join and not string formatting for this

File: relay/sources/relayd/tests/files/config/main.conf [hunk 4]
```diff
@@ -38,3 +38,6 @@ verify_certificates = false
 [remote_run]
 command = "tests/fake_agent.sh"
 use_sudo = false
+
+[shared_files_path]
+path = ""
```
Comment (amousset) [inline relay/sources/relayd/tests/files/config/main.conf]: We should put test files in `tests` to avoid cluttering the project root.

**Comment by ArmandSy** [comment]
Commit modified

**Comment by ArmandSy** [comment]
Commit modified

**Comment by ArmandSy** [comment]
Commit modified

File: relay/sources/relayd/tools/config/main.conf [hunk 5]
```diff
@@ -56,3 +56,5 @@ verify_certificates = true
 command = "/opt/rudder/bin/rudder"
 use_sudo = true
 
+[shared_files_path]
```
Comment (amousset) [inline relay/sources/relayd/tools/config/main.conf]: The section shoule probably be named `shared_files` as it may contains other information

**Comment by ArmandSy** [comment]
Commit modified

File: relay/sources/relayd/src/configuration/main.rs [hunk 6]
```diff
@@ -146,6 +147,11 @@ pub struct RemoteRun {
     pub use_sudo: bool,
 }
 
+#[derive(Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct SharedFile {
```
Comment (amousset) [inline relay/sources/relayd/src/configuration/main.rs]: `SharedFiles`

**Comment by ArmandSy** [comment]
Commit modified

File: relay/sources/relayd/src/shared_files.rs (hunk 1 above)
Comment (ncharles) [reply relay/sources/relayd/src/shared_files.rs]: so we need a new entry in tml file

**Pull request merged**